platform/pc/boot/stage1.s

```diff
@@ -95,16 +95,16 @@ loop:
 %ifdef DEBUG
         jc sector_read_error
 %endif
-        sub eax, [dap.sector_count]
-        cmp eax, 0
+        sub ax, [dap.sector_count]
+        cmp ax, 0
         jle done 
-        mov ecx, [dap.sector_count]
+        mov cx, [dap.sector_count]
         imul ecx, sectorsize
-        add [dap.offset], ecx
-        mov ecx, 0x80
-        cmp eax, ecx
-        cmovl ecx, eax 
-        mov [dap.sector_count], ecx
+        add [dap.offset], cx
```

Comment:
use ecx in this add - a sector_count of 0x80 would yield 2^16 in ecx or 0 in cx

Reply:
Right, thanks for pointing that out. I forgot to mention this earlier, but I verified that this booted properly with stage2 image sizes of 66KB and 128KB, respectively, by changing the bs flag for the dd command to those values. I couldn't figure out how to see the qemu trace- I appended the ones you mentioned to `QEMU_FLAGS` in `platform/pc/Makefile` but the output file was empty.

Reply:
> I couldn't figure out how to see the qemu trace- I appended the ones you mentioned to `QEMU_FLAGS` in `platform/pc/Makefile` but the output file was empty.

You'll need to run in emulated (TCG) mode; try making "run-noaccel" or remove --enable-kvm from the qemu command line.